File: funcscript/parser/syntax/FuncScriptParser.GetInfixExpression.cs
using FuncScript.Block;
using FuncScript.Funcs.Math;
using FuncScript.Model;

namespace FuncScript.Core
{
    public partial class FuncScriptParser
    {
        static ExpressionBlockResult GetInfixExpression(ParseContext context, int index)
        {
            var result = GetInfixExpressionSingleLevel(context, s_operatorSymbols.Length - 1, s_operatorSymbols[^1], index);

            // Removed the call to SetContext as it is no longer needed
            return result;
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.GetInfixExpression.ts
import { ParseContext, s_operatorSymbols } from "../FuncScriptParser.Main";
import { GetInfixExpressionSingleLevel } from "./FuncScriptParser.GetInfixExpressionSingleLevel";

export function GetInfixExpression(context: ParseContext, index: number) {
    const result = GetInfixExpressionSingleLevel(context, s_operatorSymbols.length - 1, s_operatorSymbols[s_operatorSymbols.length - 1], index);

    return result;
}